pkg/helper/log/pipeline_reader.go

```diff
@@ -53,21 +52,22 @@ func (r *Reader) readLivePipelineLogs(pr *v1alpha1.PipelineRun) (<-chan Log, <-c
 		trC := prTracker.Monitor(r.tasks)
 
 		wg := sync.WaitGroup{}
-		taskIndex := int32(1)
+		taskIndex := 0
 
 		for trs := range trC {
 			wg.Add(len(trs))
 
 			for _, run := range trs {
+				taskIndex++
 				// NOTE: passing tr, taskIdx to avoid data race
-				go func(tr trh.Run, taskNum int32) {
+				go func(tr trh.Run, taskNum int) {
 					defer wg.Done()
 
 					// clone the object to keep task number and name separately
 					c := r.clone()
 					c.setUpTask(int(taskNum), tr)
 					c.pipeLogs(logC, errC)
-				}(run, atomic.AddInt32(&taskIndex, 1))
+				}(run, taskIndex)
```

Comment:
Task index is passing as a argument. so it is no need using ```atomic.AddInt32```
(because not count up in goroutine)